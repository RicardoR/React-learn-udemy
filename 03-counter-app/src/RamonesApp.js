import React from 'react';

// Functional Component
const RamonesApp = () => {
    return (
      //<> </> ==> <Fragment></Fragment>
    <>
      <h1>Hey Ho! Let's go!</h1>
      <p>They're forming in straight line</p>
      <p>They're going through a tight wind</p>
      <p>The kids are losing their minds</p>
      <p>The blitzkrieg bop</p>
    </>
  );
};

export default RamonesApp;
