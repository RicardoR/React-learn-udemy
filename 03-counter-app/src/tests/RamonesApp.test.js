// import { render } from '@testing-library/react';

import React from 'react';
import RamonesApp from '../RamonesApp';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

describe('<RamonesApp /> UT', () => {
  test('Should display Hey Ho! Lets go! in the title', () => {
    const title = "Hey Ho! Let's go!";
    const ramones = {
      band: 'Los Chichos!!',
      origin: 'Los Angeles, California',
    };

    const expectedTextByData = `${ramones.band} - ${ramones.origin} - Punk!`;

    // Following code works whit import '@testing-library/jest-dom/extend-expect';
    // const wrapper = render(<RamonesApp ramones={ramones} />);
    // expect(wrapper.getByText(title)).toBeInTheDocument();
    // expect(wrapper.getByText(expectedTextByData)).toBeInTheDocument();

    const wrapper = shallow(<RamonesApp ramones={ramones} />);
    expect(wrapper.find('h1').text()).toBe(title);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').at(4).text()).toBe(expectedTextByData);
    console.log(wrapper.find('p').last().text());
    console.log(wrapper.find('p').at(4).text());
  });
});
