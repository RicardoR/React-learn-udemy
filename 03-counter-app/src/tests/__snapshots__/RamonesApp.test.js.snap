// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RamonesApp /> UT Should display Hey Ho! Lets go! in the title 1`] = `
<Fragment>
  <h1>
    Hey Ho! Let's go!
  </h1>
  <p>
    They're forming in straight line
  </p>
  <p>
    They're going through a tight wind
  </p>
  <p>
    The kids are losing their minds
  </p>
  <p>
    The blitzkrieg bop
  </p>
  <p>
    Los Chichos!!
     - 
    Los Angeles, California
     - 
    Punk!
  </p>
</Fragment>
`;
